### PR TITLE
Make RETSError a real Exception

### PR DESCRIPTION
**Pull request by arielallon**
- Needed so it can be passed as a Throwable to another exception as a $previous

### CODE REVIEW

**Review by mucha55 — CHANGES_REQUESTED** [review]
Is there a reason to remove those public methods? Even if we don't use them, we could eventually want to get our changes into PHRETS proper

**Comment by arielallon** [comment]
yea normally i wouldn't bother changing something existing in a library even if it was extraneous.
but they're defined by `\Exception` as `final`, so i can't leave them if i wanted to: https://www.php.net/manual/en/language.exceptions.extending.php

**Review by mucha55 — APPROVED** [review]

File: src/Models/RETSError.php [hunk 1]
```diff
@@ -1,18 +1,10 @@
 <?php namespace PHRETS\Models;
 
-class RETSError
+class RETSError extends \Exception
 {
     protected $code;
     protected $message;
```
Comment (jpmarcotte) [inline src/Models/RETSError.php:7]: *potential blocker:* Any reason to keep these around? Feels like a pit of failure.
```suggestion
```

**Review by jpmarcotte — APPROVED** [review]
Tentative approval.

File: src/Models/RETSError.php (hunk 1 above)
Comment (arielallon) [reply src/Models/RETSError.php:7]: Yea, trying to minimize how much of the original library I modify.
As @mucha55 noted above
> we could eventually want to get our changes into PHRETS proper

Comment (arielallon) [reply src/Models/RETSError.php:7]: Which isn't to say removing those and further modifying this class would preclude that, but the goal of this PR was to make this implement `\Throwable`. I aimed to do that as minimally-invasively as possible.

**Pull request merged**